### PR TITLE
Refactors the chef-vault HWRP and adds environment attribute.

### PR DESCRIPTION
**Pull request by johnbellone**
/cc @schisamo @someara 

Adds support for an LWRP `chef_vault_environment_secret` which wraps data the name of the environment within the actual encrypted item. It also adds appropriate helper method for accessing such data.

### CODE REVIEW

**Comment by jtimberman** [comment]
Hey @johnbellone, Sorry for the lack of response here. I'll have some inline comments, b ut as a more general comment you probably should rebase against master, as it has a new module for the chef_vault_item method, and we should use that namespace for other helpers, and include them appropriately. More background in #4.

**Comment by johnbellone** [comment]
@jtimberman Should this have a separate namespace from ChefVaultItem? After the rebase it looks like it should work the same.

**Comment by johnbellone** [comment]
@jtimberman ping

**Comment by bleything** [comment]
Oooo yes I like this. We do things a little differently and I was about to put in my own PR but maybe there's some way we can combine forces?

In our environment we separate environments into different vault items, so we can scope their searches to just the nodes that need 'em. So instead of having a vault item called `passwords` like this:

```json
{
  "production": "prd value",
  "staging": "stg value",
}
```

We have two vaults called `passwords-production` and `passwords-staging` with a single value in each.

What would y'all think about supporting both modes? I can think a bit on how to accomplish that in a clean way.

**Comment by jtimberman** [comment]
Hey @johnbellone - Sorry I was on vacation and then at ChefConf, this slipped off my radar.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+language: ruby
+script: bin/rspec
+rvm:
+  - 1.9.3
```
Comment (jtimberman) [inline .travis.yml]: 1.9.3 is now [EOL](https://www.ruby-lang.org/en/news/2014/01/10/ruby-1-9-3-will-end-on-2015/).

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  - 2.1
+branches:
+  only:
+    - master
+builder_args: --jobs 7
+notifications:
+  irc:
+    channels:
+      - 'chat.freenode.net#chef'
```
Comment (jtimberman) [inline .travis.yml]: We don't do any notifications to IRC for other cookbooks, this should be removed.

**Comment by jtimberman** [comment]
It would be super great to have mock data bag items that can be used like the stuff in the test/integration directory, and Serverspec tests.

**Comment by johnbellone** [comment]
@jtimberman See those new commits.

**Comment by jtimberman** [comment]
Looks good. I left a comment on 6633121 instead of in this PR because I'm bad at using a web browser :sweat_smile:.

Do you think you'd be able to get the fixtures data into the test directory and write that into a recipe for testing this that can be validated with serverspec?

**Comment by johnbellone** [comment]
This would close #31 and #32.

**Comment by johnbellone** [comment]
Thoughts?

**Comment by pburkholder** [comment]
:+1:

**Comment by johnbellone** [comment]
@jtimberman

**Comment by irvingpop** [comment]
hey @johnbellone, this is almost ready to merge.   rspec passes, that's good.   

However I'm finding serverspec fails because the node sees itself in the `_default` environment, not the test environment:
```    
         1) File "/tmp/chef-vault-environment-secret" content should match /unicorn/
            Failure/Error: its(:content) { should match(/unicorn/) }
              expected "" to match /unicorn/
              Diff:
              @@ -1,2 +1,2 @@
              -/unicorn/
              +""
              
            # /tmp/verifier/suites/serverspec/default_spec.rb:8:in `block (2 levels) in <top (required)>'
       
         2) File "/tmp/chef-vault-environment-secret-from-library" content should match /unicorn/
            Failure/Error: its(:content) { should match(/unicorn/) }
              expected "" to match /unicorn/
              Diff:
              @@ -1,2 +1,2 @@
              -/unicorn/
              +""
              
            # /tmp/verifier/suites/serverspec/default_spec.rb:20:in `block (2 levels) in <top (required)>'       
```

I fixed this locally by adding a `test.json` environment to `test/fixtures/environments` and made the following tweak to `.kitchen.yml`:

```diff
diff --git a/.kitchen.yml b/.kitchen.yml
index be6d6f7..e211a35 100644
--- a/.kitchen.yml
+++ b/.kitchen.yml
@@ -6,7 +6,10 @@ provisioner:
   name: chef_zero
   nodes_path: test/fixtures/nodes
   clients_path: test/fixtures/clients
+  environments_path: test/fixtures/environments
   data_bags_path: test/fixtures/data_bags
+  client_rb:
+    environment: test
 
 platforms:
   - name: ubuntu-14.04
```

Optional, but I also added something to the `secrets/bacon.json` to make debugging easier:
```diff
--- a/test/fixtures/data_bags/secrets/bacon.json
+++ b/test/fixtures/data_bags/secrets/bacon.json
@@ -1,5 +1,8 @@
 {
     "id": "bacon",
+    "_default": {
+      "type": "unikitten"
+    },
     "test": {
         "type": "unicorn"
     }

```

Could you please add that to your PR and update the description so it no longer says "work-in-progress" ?

**Comment by johnbellone** [comment]
@irvingpop it should be good now.

**Comment by irvingpop** [comment]
Thank you @johnbellone !!

**Pull request merged**